src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -100,6 +100,23 @@ public bool OnUpdateTopology(Topology topology, bool forceUpdate = false)
             return UnlikelyEveryoneFaultedChoice(state);
         }
 
+        public (int Index, ServerNode Node, long TopologyEtag) GetPreferredNodeWithTopology()
```

Comment:
Isn't this duplicating the code in `GetPreferredNode`? Can we merge these?

Reply:
done